### ISSUE TITLE
Making package ST3 only

### ISSUE THREAD

**Issue by jrappen** [aziz/DistractionFreeWindow#22]
@aziz I was looking at wbond/packagecontrol.io#90 recently and reviewing the code of this repository it seems to me like most of the code here is workarounds for ST2.

Moving forward I would suggest:

* make the package ST3 only
  * we should be able to get rid of the package's settings file and run all functionality off the application's `User/Preferences.sublime-settings` and `User/Distraction Free.sublime-settings` files only
    * add a python file during the transition phase to auto-migrate settings according to expected behaviour
  * add a `dependencies.json` to show users the README and CHANGELOG with ST3's mini-html via [mdpopups](https://github.com/facelessuser/sublime-markdown-popups)

Your decision ...

**Comment by jrappen** [comment]
## Minimal setup on ST3

### `distraction_free_window.py`

```python
#!/usr/bin/env python
# coding: utf-8

import sublime
import sublime_plugin


class DistractionFreeWindowCommand(sublime_plugin.WindowCommand):

    def run(self):
        w = self.window
        if w is None:
            status_msg('Distraction Free Window: Error: Window is None')
            return

        if w.active_view().settings().get('is_widget', False):
            status_msg('Distraction Free Window: Error: Active view is a widget.')
            return

        if w.is_sidebar_visible():
            # Preferences > Settings - Distraction Free
            distraction_free = sublime.load_settings('Distraction Free.sublime-settings')
            w.settings().set('draw_centered', distraction_free.get('draw_centered', True))
            w.settings().set('draw_indent_guides', distraction_free.get('draw_indent_guides', True))
            w.settings().set('draw_white_space', distraction_free.get('draw_white_space', 'selection'))
            w.settings().set('fold_buttons', distraction_free.get('fold_buttons', True))
            w.settings().set('gutter', distraction_free.get('gutter', False))
            w.settings().set('line_numbers', distraction_free.get('line_numbers', False))
            w.settings().set('rulers', distraction_free.get('rulers', []))
            w.settings().set('scroll_past_end', distraction_free.get('scroll_past_end', True))
            w.settings().set('word_wrap', distraction_free.get('word_wrap', True))
            w.settings().set('wrap_width', distraction_free.get('wrap_width', 80))
            if sublime.platform() is not 'osx':
                w.set_menu_visible(False)
            w.set_sidebar_visible(False)
            w.set_tabs_visible(False)
            w.set_minimap_visible(False)
            # w.set_status_bar_visible(False)
        else:
            # Preferences > Settings
            preferences = sublime.load_settings('Preferences.sublime-settings')
            w.settings().set('draw_centered', preferences.get('draw_centered', False))
            w.settings().set('draw_indent_guides', preferences.get('draw_indent_guides', True))
            w.settings().set('draw_white_space', preferences.get('draw_white_space', 'selection'))
            w.settings().set('fold_buttons', preferences.get('fold_buttons', True))
            w.settings().set('gutter', preferences.get('gutter', True))
            w.settings().set('line_numbers', preferences.get('line_numbers', True))
            w.settings().set('rulers', preferences.get('rulers', []))
            w.settings().set('scroll_past_end', preferences.get('scroll_past_end', True))
            w.settings().set('word_wrap', preferences.get('word_wrap', 'auto'))
            w.settings().set('wrap_width', preferences.get('wrap_width', 0))
            if sublime.platform() is not 'osx':
                w.set_menu_visible(True)
            w.set_sidebar_visible(True)
            w.set_tabs_visible(True)
            w.set_minimap_visible(True)
            # w.set_status_bar_visible(True)
```

### `Default.sublime-keymap`

```json
[
    { "keys": ["super+f11"], "command": "distraction_free_window" }
]
```

**Comment by jrappen** [comment]
@aziz Currently Sublime Text 3 users make up 97.4% of the Package Control activity. Should I open a PR with the changes suggested above?

**Comment by jrappen** [comment]
Also, I'd like your take on using 3 sets of preferences vs using 2 sets of preferences.

Would you prefer:

* Preferences
* Distraction Free
* Distraction Free Window

Or rather?

* Preferences
* Distraction Free (for both Distraction Free and Distraction Free Window)

**Comment by aziz** [comment]
> @aziz Currently Sublime Text 3 users make up 97.4% of the Package Control activity. Should I open a PR with the changes suggested above?

I think we can move the current version as is into a branch and then release a new one just for version 3.

**Comment by aziz** [comment]
> Also, I'd like your take on using 3 sets of preferences vs using 2 sets of preferences.

I'm not sold on this one. what's the benefit of it for the user? it'll just make things more complicated. 
given that we can use comments and new lines to separate the setting into sections, I think I prefer less setting files

**Comment by jrappen** [comment]
well currently you're using 3 settings files:

* these two: <https://github.com/aziz/DistractionFreeWindow/tree/master/settings>
* plus preferences

**Comment by jrappen** [comment]
It seems we can separate by branches anyhow, but PC uses tag prefixes. The reference on the package_control_channel repo would need an update, too...

---

Compare `BracketHighlighter`:

* <https://github.com/facelessuser/BracketHighlighter/releases>
* <https://github.com/wbond/package_control_channel/blob/c0710cf8841182293b8aa2c3acf5df06d5973156/repository/b.json#L1153-L1165>

**Comment by aziz** [comment]
Sorry I got it totally wrong. I'm neutral. given the fact that the `Distraction Free` settings that we added is something internal that users should not change. it's only for distinguishing between DFW and the real DF mode in sublime. but if you feel it makes it easier, feel free to get rid of a separate setting and put everything under the DF settings. make sure it's future proof and we'll not have any potential conflict with future native DF settings.

**Comment by jrappen** [comment]
Just asking because my understanding from the introduction in the README is that you built the package to have a non-fullscreen Distraction Free mode....implying that the non-fullscreen part is the **only** difference. Logically the next step for me would be the removal of the third set of settings.

I'm just documenting my thoughts here, I'm neutral as well.

**Comment by jrappen** [comment]
@aziz Released [`v0.6.7`](https://github.com/aziz/DistractionFreeWindow/releases/tag/0.6.7) with a **request for comment**. I suggest waiting a week or so, then splitting branches and releases.

**Comment by jrappen** [comment]
@aziz I split branches locally, removed redundant code where applicable, prepared with prefixed tags. Will push tonight after work and send a pull request to the package control channel with an updated config.

**Comment by jrappen** [comment]
Split into branches:

* [`master`](https://github.com/aziz/DistractionFreeWindow/tree/master)
* [`st2`](https://github.com/aziz/DistractionFreeWindow/tree/st2)

to apply [tags](https://github.com/aziz/DistractionFreeWindow/tags) with the following prefixes:

* `st3-`
* `st2-`

going forward.

---

Fixed in wbond/package_control_channel#7153
